docs/debugging.qmd

```diff
@@ -0,0 +1,144 @@
+[in the `la-metro-councilmatic` Sentry project](https://sentry.io/organizations/datamade/issues/?project=2131912),
+however sometimes steps run without failing but don't generate the desired result.
+Read on for more on each step of the pipeline, plus past failures and their resolutions.
+
+#### `refresh_pic`
+
+**Where it lives:** [Django Councilmatic](https://github.com/datamade/django-councilmatic/blob/2.5/councilmatic_core/management/commands/refresh_pic.py)<br />
+**What is does:** Deletes [cached documents](https://github.com/datamade/property-image-cache) for recently updated bills and events
+
+**Past issues:**
+
+- [Cached event agenda was out of sync with Legistar](https://github.com/datamade/la-metro-councilmatic/issues/443).
+We have since updated the logic for which documents to remove from the cache, so this error should be resolved,
+but the linked issue contains instructions for resolving this error manually, in case we see a regression.
+
+#### `compile_pdfs`
+
+**Where it lives:** [LA Metro Councilmatic](https://github.com/datamade/la-metro-councilmatic/blob/main/lametro/management/commands/compile_pdfs.py)<br />
+**What it does:** Notifies the [`metro-pdf-merger`](https://github.com/datamade/metro-pdf-merger) of new documents that need to be merged into a bill or event packet
+
+**Past issues:**
+
+- [Sometimes the worker fails to merge documents, resulting in missing packets](https://github.com/datamade/la-metro-councilmatic/issues/476).
+There should be a corresponding error for this [in the `metro-pdf-merger` Sentry project](https://sentry.io/organizations/datamade/issues/?project=155211),
+however the project is pretty noisy, so you can shell into the server (`metro-pdf-merger.datamade.us`) and tail or grep the worker logs to double check.
+- [The worker PDF merger has died mysteriously](https://github.com/datamade/metro-pdf-merger/issues/19).
+
+#### `convert_attachment_text`
+
+**Where it lives:** [Django Councilmatic](https://github.com/datamade/django-councilmatic/blob/2.5/councilmatic_core/management/commands/convert_attachment_text.py)<br />
+**What it does:** Extracts text from bill attachments for indexing
+
+**Past issues:**
+
+N/A
+
+#### `update_index`
+
+**Where it lives:** [Haystack](https://django-haystack.readthedocs.io/en/master/management_commands.html#update-index)<br />
+**What it does:** Updates the search index
+
+**Past issues:**
+
+- [Failures to update the index fail silently](https://github.com/datamade/la-metro-councilmatic/issues/535).
+We have updated the Haystack config to raise Solr errors immediately, and we've also increased the heap space allowed to the production Solr instance,
+so this should be taken care of. If Solr errors recur, and you see log messages about heap space,
+shell into the Councilmatic server and restart the production Solr instance: `sudo docker restart lametro-production-solr`.
+Then, switch over to the application directory, and rebuild the index!
```

Comment:
I'm inclined to just remove this entry since we're on Elastic now. I don't believe we've run into this particular error so far. Also, we're running a managed Elastic instance anyway so there's no analog for that restart command.